hydra_redis/querying_mechanism.py

```diff
@@ -354,6 +354,69 @@ def faceted_key(self, key, value):
         """
         return ("{}".format("fs:" + key + ":" + value))
 
+    def convert_byte_string(self, value_set):
+        """
+        It converts byte strings to strings.
+        """
+        new_value_set = set()
+        for obj in value_set:
+            string = obj.decode('utf-8')
+            new_value_set.add(string)
+        return new_value_set
+
+    def and_or_query(self, query_list):
+        """
+        It is a recursive function.
+        It takes the arguement as list(query_list)
+        which contains the faceted indexes and operation and brackets also.
+        List Ex:['fs:model:xyz', 'and', '(', 'fs:name:Drone1', 'or',
+                'fs:name:Drone2', ')']
+                for query "model xyz and (name Drone1 or name Drone2)"
+        :param query_list: get a list of faceted indexes and operations
+        :param return: get data from the Redis memory for specific query.
+        """
+        # check if there is both "and" and "or" with help of bracket.
+        if ")" not in query_list:
```

Comment:
What about the precedence thing we talked about?

Reply:
I'll work on that. Now I am trying to make Dockerfile. And I think brackets are good to use because it makes query simpler and easily understandable.
Because using brackets a simple query can asked in many diff. ways like:
`a and b and c or d or e and f`
it can be like these:
`(a and b and c) or d or (e and f)` and
`a and b and (c or d or e) and f` and 
`a and b and ((c or d or e) and f`

What do you think?

Reply:
Yeah, I agree. There's no question the format with brackets is better and more understandable but supporting queries without brackets would also be cool. It's just a add-on sort of thing to support a wider format of queries.

Reply:
Okay, Can I update it later? Because I think it is not a high priority issue like others.
After some other issues, I'll update it.